src/pages/api/signup.ts

```diff
@@ -0,0 +1,125 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { connectToDatabase } from '../utils/db';
+import User, { IUser } from '../schemas/user.schema';
+import { sendOTPEmail } from '../helper/sendOTP';
+import { generateOTP } from '../helper/generateOTP';
+
+export type SignUpResponse = {
+  message: string;
+};
+
+export type UserInput = {
+  firstName: string;
+  lastName: string;
+  email: string;
+};
+
+let isConnected = false;
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<SignUpResponse>
+) {
+  if (req.method === 'POST') {
+    try {
+      // Ensure database connection
+      if (!isConnected) {
+        await connectToDatabase();
+        isConnected = true;
+      }
+
+      const { firstName, lastName, email } = req.body as UserInput;
+
+      // Basic validation
+      if (!firstName || !lastName || !email) {
+        return res.status(400).json({ message: 'All fields are required' });
+      }
+      console.log(email);
```

Comment:
never do console like this. always add text with console. or else u will urself get confused from where is this console coming.